docs/docs/dev-setup.md

```diff
@@ -4,9 +4,189 @@ title: Basic Setup
+	sh "zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run" --quiet -- -d /opt/toolchains/zephyr-sdk-${ZSDK_VERSION} && \
+	rm "zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run"
+```
+
+The installation will prompt with several questions about installation location, and creating a default `~/.zephyrrc` for you with various variables. The defaults shouldn normally work as expected.
+
+### Source Code
+
+Next, you'll need to clone the ZMK source repository if you haven't already:
+
+```
+git clone https://github.com/zmkfirmware/zmk.git
+```
+
+### Initialize & Update Zephy Workspace
+
+Since ZMK is built as a Zephyr™ application, the next step is
+to use `west` to initialize and update your workspace. The ZMK
+Zephyr™ application is in the `app/` source directory:
+
+#### Initialize West
+
+```bash
+west init -l app/
+```
+
+#### Update To Fetch Modules
+
+```bash
+west update
+```
+
+#### Export Zephyr™ Core
+
+```bash
+west zephyr-export
+```
+
+### Environment Variables
+
+By default, the Zephyr™ SDK will create a file named `~/.zephyrrc` with the correct environment variables to build ZMK.
+We suggest two main [options](https://docs.zephyrproject.org/latest/guides/env_vars.html?highlight=zephyrrc) for how to load those settings.
+
+#### Per Shell
+
+To load the Zephyr environment properly for just one transient shell, run:
+
+```
+source zmk/zephyr/zephyr-env.sh
+```
+
+#### All Shells
+
+To load the environment variables for your shell every time,
+append the existing `~/.zephyrrc` file to your shell's RC file, e.g.
+
+##### Bash
+
+```
+cat ~/.zephyrc >> ~/.bashrc
+```
+
+##### ZSH
+
+```
+cat ~/.zephyrc >> ~/.zshrc
+```
+
+## Build
+
+Actually building the ZMK firmware occurs within the `app/` subdirectory
+of the ZMK repository. To build for your particular keyboard, the behaviour varies slightly depending on if you are building for a keyboard with
+an onboard MCU, or one that uses a MCU board addon.
+
+### Keyboard (Shield) + MCU Board
+
+ZMK treats keyboards that take a MCU addon board as [shields](https://docs.zephyrproject.org/latest/guides/porting/shields.html), and treats the smaller MCU board as the true [board](https://docs.zephyrproject.org/latest/guides/porting/board_porting.html)
+
+Given the following:
+
+- MCU Board: Proton-C
+- Keyboard PCB: kyria
+- Keymap: default
+
+You can build ZMK with the following:
+
+```bash
+west build -b proton_c -- -DSHIELD=kyria -DKEYMAP=default
+```
+
+### Keyboard With Onboard MCU
+
+Keyboards with onboard MCU chips are simply treated as the [board](https://docs.zephyrproject.org/latest/guides/porting/board_porting.html) as far as Zephyr™ is concerned.
+
+Given the following:
+
+- Keyboard: Planck
+- Keymap: default
+
+you can build ZMK with the following:
+
+```bash
+west build -b planck -- -DKEYMAP=default
+```
+
+## Flashing
+
+Once built, the previously supplied parameters will be remember, so you can simply run the following to flash your
+board, with it in bootloader mode:
+
+```
+west flash
```

Comment:
I'm not sure if this will work in WSL with openocd... Not sure if it can see external flashing tools like ST-Link. I can't tell if WSL can't find my ST-Link or if it can't open a connection with the board (because I don't have the board). I just get:

```bash
Info : clock speed 500kHz
Error : open failed
```

Same error with or without the ST-Link installed, so I'd say it just can't find the ST-Link.

Maybe should check this out: https://gojimmypi.blogspot.com/2017/04/installing-openocd-on-windows-wsl-ubuntu.html